app/scripts/metamask-controller.test.js

```diff
@@ -180,8 +181,10 @@ describe('MetaMaskController', function () {
   before(async function () {
     globalThis.isFirstTimeProfileLoaded = true;
     await ganacheServer.start();
-    sinon.spy(MetaMaskController.prototype, 'resetStates');
-    sinon.spy(MetaMaskControllerMV3.prototype, 'resetStates');
+    resetStateSpies = [
+      sinon.spy(MetaMaskController.prototype, 'resetStates'),
```

Comment:
Typically our strategy for keeping tests isolated is to construct spies/mocks/fakes fresh for each test, then discard them at the end by calling `sinon.restore()`. Would this work here as well?

That would look like moving these two spies to `beforeEach`.

And for the cleanup step, we could add `sinon.restore` to `afterEach`, or we could switch these two mocks to use `sandbox.spy` rather than `sinon.spy` so they're covered by the pre-existing `sandbox.restore` in `beforeEach` already. (ultimately we should be using the default sandbox here, but fixing that is a bigger problem).

Reply:
Seems like a good idea to me! It's not so hard to review with "hide whitespace" toggled